# C:\Users\coren\AppData\Local\opam\default\bin\dune.exe runtest tests/type/sans_fonction/
# OCAMLPARAM: unset
# Shared cache: disabled
# Shared cache location:
# C:\Users\coren\AppData\Local\Microsoft\Windows\INetCache\dune\db
# Workspace root: "C:\\Users\\coren\\Documents\\Projets\\Projet_TDL"
# Auto-detected concurrency: 16
# Dune context:
#  { name = "default"
#  ; kind = "default"
#  ; profile = Dev
#  ; merlin = true
#  ; fdo_target_exe = None
#  ; build_dir = In_build_dir "default"
#  ; instrument_with = []
#  }
$ C:\Users\coren\AppData\Local\opam\default\bin\ocamlc.opt.exe -config > C:\Users\coren\AppData\Local\Temp\dune_abc1f8_output
$ (cd _build/default && .ppx/423b3c0918a91461d11b9cd1259aca91/ppx.exe --cookie "inline_tests=\"enabled\"" --cookie "library-name=\"rat\"" -o passeTypeRat.pp.ml --impl passeTypeRat.ml -corrected-suffix .ppx-corrected -diff-cmd - -dump-ast)
$ (cd _build/default && C:\Users\coren\AppData\Local\opam\default\bin\ocamldep.opt.exe -modules -impl passeTypeRat.pp.ml) > _build/default/.rat.objs/rat__PasseTypeRat.impl.d
$ (cd _build/default && C:\Users\coren\AppData\Local\opam\default\bin\ocamlc.opt.exe -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -warn-error -A -g -bin-annot -bin-annot-occurrences -I .rat.objs/byte -I C:/Users/coren/AppData/Local/opam/default/lib/ocaml\unix -I C:\Users\coren\AppData\Local\opam\default\lib\base -I C:\Users\coren\AppData\Local\opam\default\lib\base\base_internalhash_types -I C:\Users\coren\AppData\Local\opam\default\lib\base\shadow_stdlib -I C:\Users\coren\AppData\Local\opam\default\lib\jane-street-headers -I C:\Users\coren\AppData\Local\opam\default\lib\ocaml_intrinsics_kernel -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_compare\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_enumerate\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\config -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\config_types -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\make_corrected_file -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\runtime -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_hash\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_here\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_inline_test\config -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_inline_test\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_sexp_conv\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppxlib\print_diff -I C:\Users\coren\AppData\Local\opam\default\lib\sexplib0 -I C:\Users\coren\AppData\Local\opam\default\lib\stdio -I C:\Users\coren\AppData\Local\opam\default\lib\time_now -no-alias-deps -opaque -open Rat -o .rat.objs/byte/rat__PasseTypeRat.cmo -c -impl passeTypeRat.pp.ml)
> File "passeTypeRat.ml", line 4, characters 0-15:
> 4 | open Exceptions
>     ^^^^^^^^^^^^^^^
> Warning 33 [unused-open]: unused open Rat.Exceptions.
>
> File "passeTypeRat.ml", line 18, characters 23-27:
> 18 | | AstTds.AppelFonction(info, le) -> (AstType.Booleen(true) , Bool)
>                             ^^^^
> Warning 27 [unused-var-strict]: unused variable info.
>
> File "passeTypeRat.ml", line 18, characters 29-31:
> 18 | | AstTds.AppelFonction(info, le) -> (AstType.Booleen(true) , Bool)
>                                   ^^
> Warning 27 [unused-var-strict]: unused variable le.
$ (cd _build/default && C:\Users\coren\AppData\Local\opam\default\bin\ocamlopt.opt.exe -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -warn-error -A -g -I .rat.objs/byte -I .rat.objs/native -I C:/Users/coren/AppData/Local/opam/default/lib/ocaml\unix -I C:\Users\coren\AppData\Local\opam\default\lib\base -I C:\Users\coren\AppData\Local\opam\default\lib\base\base_internalhash_types -I C:\Users\coren\AppData\Local\opam\default\lib\base\shadow_stdlib -I C:\Users\coren\AppData\Local\opam\default\lib\jane-street-headers -I C:\Users\coren\AppData\Local\opam\default\lib\ocaml_intrinsics_kernel -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_compare\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_enumerate\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\config -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\config_types -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\make_corrected_file -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\runtime -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_hash\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_here\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_inline_test\config -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_inline_test\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_sexp_conv\runtime-lib -I C:\Users\coren\AppData\Local\opam\default\lib\ppxlib\print_diff -I C:\Users\coren\AppData\Local\opam\default\lib\sexplib0 -I C:\Users\coren\AppData\Local\opam\default\lib\stdio -I C:\Users\coren\AppData\Local\opam\default\lib\time_now -intf-suffix .ml -no-alias-deps -opaque -open Rat -o .rat.objs/native/rat__PasseTypeRat.cmx -c -impl passeTypeRat.pp.ml)
> File "passeTypeRat.ml", line 4, characters 0-15:
> 4 | open Exceptions
>     ^^^^^^^^^^^^^^^
> Warning 33 [unused-open]: unused open Rat.Exceptions.
>
> File "passeTypeRat.ml", line 18, characters 23-27:
> 18 | | AstTds.AppelFonction(info, le) -> (AstType.Booleen(true) , Bool)
>                             ^^^^
> Warning 27 [unused-var-strict]: unused variable info.
>
> File "passeTypeRat.ml", line 18, characters 29-31:
> 18 | | AstTds.AppelFonction(info, le) -> (AstType.Booleen(true) , Bool)
>                                   ^^
> Warning 27 [unused-var-strict]: unused variable le.
$ (cd _build/default && C:\Users\coren\AppData\Local\opam\default\bin\ocamlopt.opt.exe -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -warn-error -A -g -a -o rat.cmxa .rat.objs/native/rat.cmx .rat.objs/native/rat__Type.cmx .rat.objs/native/rat__Tds.cmx .rat.objs/native/rat__Tam.cmx .rat.objs/native/rat__Ast.cmx .rat.objs/native/rat__PrinterAst.cmx .rat.objs/native/rat__Exceptions.cmx .rat.objs/native/rat__PasseTypeRat.cmx .rat.objs/native/rat__PasseTdsRat.cmx .rat.objs/native/rat__Passe.cmx .rat.objs/native/rat__Parser.cmx .rat.objs/native/rat__Lexer.cmx .rat.objs/native/rat__Compilateur.cmx .rat.objs/native/rat__Code.cmx)
$ (cd _build/default && C:\Users\coren\AppData\Local\opam\default\bin\ocamlopt.opt.exe -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -warn-error -A -w -24 -g -g -o tests/type/sans_fonction/.sans_fonction_type.inline-tests/inline_test_runner_sans_fonction_type.exe C:/Users/coren/AppData/Local/opam/default/lib/ocaml\unix\unix.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_inline_test\config/inline_test_config.cmxa C:\Users\coren\AppData\Local\opam\default\lib\base\base_internalhash_types/base_internalhash_types.cmxa -I C:\Users\coren\AppData\Local\opam\default\lib\base\base_internalhash_types C:\Users\coren\AppData\Local\opam\default\lib\sexplib0\sexplib0.cmxa C:\Users\coren\AppData\Local\opam\default\lib\base\shadow_stdlib/shadow_stdlib.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ocaml_intrinsics_kernel\ocaml_intrinsics_kernel.cmxa -I C:\Users\coren\AppData\Local\opam\default\lib\ocaml_intrinsics_kernel C:\Users\coren\AppData\Local\opam\default\lib\base\base.cmxa -I C:\Users\coren\AppData\Local\opam\default\lib\base C:\Users\coren\AppData\Local\opam\default\lib\jane-street-headers\jane_street_headers.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_sexp_conv\runtime-lib/ppx_sexp_conv_lib.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_compare\runtime-lib/ppx_compare_lib.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_enumerate\runtime-lib/ppx_enumerate_lib.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_hash\runtime-lib/ppx_hash_lib.cmxa C:\Users\coren\AppData\Local\opam\default\lib\time_now\time_now.cmxa -I C:\Users\coren\AppData\Local\opam\default\lib\time_now C:\Users\coren\AppData\Local\opam\default\lib\ppx_inline_test\runtime-lib/ppx_inline_test_lib.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_here\runtime-lib/ppx_here_lib.cmxa C:\Users\coren\AppData\Local\opam\default\lib\stdio\stdio.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppxlib\print_diff/ppxlib_print_diff.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\make_corrected_file/make_corrected_file.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\config_types/expect_test_config_types.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\config/expect_test_config.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\runtime/ppx_expect_runtime.cmxa -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\runtime rat.cmxa tests/type/sans_fonction/sans_fonction_type.cmxa C:\Users\coren\AppData\Local\opam\default\lib\ppx_inline_test\runner/lib/ppx_inline_test_runner_lib.cmxa -I C:\Users\coren\AppData\Local\opam\default\lib\ppx_inline_test\runner/lib C:\Users\coren\AppData\Local\opam\default\lib\ppx_expect\evaluator/ppx_expect_evaluator.cmxa tests/type/sans_fonction/.sans_fonction_type.inline-tests/.sans_fonction_type.inline-tests.eobjs/native/dune__exe__Inline_test_runner_sans_fonction_type.cmx -linkall)
$ (cd _build/default/tests/type/sans_fonction && .sans_fonction_type.inline-tests/inline_test_runner_sans_fonction_type.exe inline-test-runner sans_fonction_type -list-partitions) > _build/default/tests/type/sans_fonction/.sans_fonction_type.inline-tests/partitions-best
$ (cd _build/default/tests/type/sans_fonction && .sans_fonction_type.inline-tests/inline_test_runner_sans_fonction_type.exe inline-test-runner sans_fonction_type -partition test.ml -source-tree-root ../../.. -diff-cmd -)
> Erreur Fraction 
> Erreur Fraction 
> Erreur PLus 
> ../../../../../tests/tam/sans_fonction/fichiersRat/complique.rat
> File "tests/type/sans_fonction/test.ml", line 69, characters 0-100: testAffectation1 threw Rat.Exceptions.TypeInattendu(1, 3).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 105, characters 10-51
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 70, characters 10-59
>
> File "tests/type/sans_fonction/test.ml", line 72, characters 0-100: testAffectation2 threw Rat.Exceptions.TypeInattendu(0, 3).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 105, characters 10-51
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 73, characters 10-59
>
> File "tests/type/sans_fonction/test.ml", line 75, characters 0-100: testAffectation3 threw Rat.Exceptions.TypeInattendu(2, 3).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 105, characters 10-51
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 76, characters 10-59
>
> File "tests/type/sans_fonction/test.ml", line 78, characters 0-174: testAffectation4 threw Rat.Exceptions.TypeInattendu(1, 3).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 105, characters 10-51
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 80, characters 12-61
>
> File "tests/type/sans_fonction/test.ml", line 85, characters 0-175: testAffectation5 threw Rat.Exceptions.TypeInattendu(0, 3).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 105, characters 10-51
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 87, characters 12-61
>
> File "tests/type/sans_fonction/test.ml", line 92, characters 0-175: testAffectation6 threw Rat.Exceptions.TypeInattendu(0, 3).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 105, characters 10-51
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 94, characters 12-61
>
> File "tests/type/sans_fonction/test.ml", line 99, characters 0-174: testAffectation7 threw Rat.Exceptions.TypeInattendu(2, 3).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 105, characters 10-51
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 101, characters 12-61
>
> File "tests/type/sans_fonction/test.ml", line 106, characters 0-175: testAffectation8 threw Rat.Exceptions.TypeInattendu(2, 3).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 105, characters 10-51
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 108, characters 12-61
>
> File "tests/type/sans_fonction/test.ml", line 113, characters 0-175: testAffectation9 threw Rat.Exceptions.TypeInattendu(1, 3).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 105, characters 10-51
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 115, characters 12-61
>
> File "tests/type/sans_fonction/test.ml", line 137, characters 0-106: testConditionnelle4 threw Rat.Exceptions.TypeInattendu(3, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 124, characters 10-53
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 138, characters 10-62
>
> File "tests/type/sans_fonction/test.ml", line 140, characters 0-181: testConditionnelle5 threw Rat.Exceptions.TypeInattendu(3, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 124, characters 10-53
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 142, characters 12-64
>
> File "tests/type/sans_fonction/test.ml", line 147, characters 0-181: testConditionnelle6 threw Rat.Exceptions.TypeInattendu(3, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 124, characters 10-53
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 149, characters 12-64
>
> File "tests/type/sans_fonction/test.ml", line 171, characters 0-98: testRepetition4 threw Rat.Exceptions.TypeInattendu(3, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 131, characters 12-55
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 172, characters 10-58
>
> File "tests/type/sans_fonction/test.ml", line 174, characters 0-173: testRepetition5 threw Rat.Exceptions.TypeInattendu(3, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 131, characters 12-55
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 176, characters 12-60
>
> File "tests/type/sans_fonction/test.ml", line 181, characters 0-173: testRepetition6 threw Rat.Exceptions.TypeInattendu(3, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 131, characters 12-55
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 183, characters 12-60
>
> File "tests/type/sans_fonction/test.ml", line 200, characters 0-96: testRationnel2 threw (Failure "Erreur interne").
>   Raised at Stdlib.failwith in file "stdlib.ml", line 29, characters 17-33
>   Called from Rat__PasseTypeRat.analyse_type_expression in file "passeTypeRat.ml", line 36, characters 49-75
>   Called from Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 84, characters 21-46
>   Called from Stdlib__List.map in file "list.ml", line 83, characters 15-19
>   Called from Stdlib__List.map in file "list.ml", line 88, characters 14-21
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 201, characters 10-57
>
> File "tests/type/sans_fonction/test.ml", line 224, characters 0-98: testNumerateur1 threw Rat.Exceptions.TypeInattendu(2, 1).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 83, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 225, characters 10-58
>
> File "tests/type/sans_fonction/test.ml", line 227, characters 0-98: testNumerateur2 threw Rat.Exceptions.TypeInattendu(3, 2).
>   Raised at Rat__PasseTypeRat.analyse_type_expression in file "passeTypeRat.ml", line 35, characters 2-43
>   Called from Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 84, characters 21-46
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 228, characters 10-58
>
> File "tests/type/sans_fonction/test.ml", line 244, characters 0-173: testNumerateur5 threw Rat.Exceptions.TypeInattendu(2, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 83, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 246, characters 12-60
>
> File "tests/type/sans_fonction/test.ml", line 251, characters 0-102: testDenominateur1 threw Rat.Exceptions.TypeInattendu(2, 1).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 83, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 252, characters 10-60
>
> File "tests/type/sans_fonction/test.ml", line 254, characters 0-102: testDenominateur2 threw Rat.Exceptions.TypeInattendu(3, 2).
>   Raised at Rat__PasseTypeRat.analyse_type_expression in file "passeTypeRat.ml", line 35, characters 2-43
>   Called from Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 84, characters 21-46
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 255, characters 10-60
>
> File "tests/type/sans_fonction/test.ml", line 271, characters 0-177: testDenominateur5 threw Rat.Exceptions.TypeInattendu(2, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 83, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 273, characters 12-62
>
> File "tests/type/sans_fonction/test.ml", line 278, characters 0-88: testIdent1 threw Rat.Exceptions.TypeInattendu(3, 1).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 279, characters 10-53
>
> File "tests/type/sans_fonction/test.ml", line 281, characters 0-88: testIdent2 threw Rat.Exceptions.TypeInattendu(3, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 282, characters 10-53
>
> File "tests/type/sans_fonction/test.ml", line 284, characters 0-88: testIdent3 threw Rat.Exceptions.TypeInattendu(3, 2).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 285, characters 10-53
>
> File "tests/type/sans_fonction/test.ml", line 287, characters 0-162: testIdent4 threw Rat.Exceptions.TypeInattendu(3, 1).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 289, characters 12-55
>
> File "tests/type/sans_fonction/test.ml", line 294, characters 0-163: testIdent5 threw Rat.Exceptions.TypeInattendu(3, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 296, characters 12-55
>
> File "tests/type/sans_fonction/test.ml", line 301, characters 0-163: testIdent6 threw Rat.Exceptions.TypeInattendu(3, 0).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 303, characters 12-55
>
> File "tests/type/sans_fonction/test.ml", line 308, characters 0-162: testIdent7 threw Rat.Exceptions.TypeInattendu(3, 2).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 310, characters 12-55
>
> File "tests/type/sans_fonction/test.ml", line 315, characters 0-163: testIdent8 threw Rat.Exceptions.TypeInattendu(3, 2).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 317, characters 12-55
>
> File "tests/type/sans_fonction/test.ml", line 322, characters 0-163: testIdent9 threw Rat.Exceptions.TypeInattendu(3, 1).
>   Raised at Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 94, characters 10-49
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 324, characters 12-55
>
> File "tests/type/sans_fonction/test.ml", line 405, characters 0-138: all_tam threw (Failure "Erreur interne").
>   Raised at Stdlib.failwith in file "stdlib.ml", line 29, characters 17-33
>   Called from Rat__PasseTypeRat.analyse_type_expression in file "passeTypeRat.ml", line 37, characters 17-43
>   Called from Rat__PasseTypeRat.analyse_type_instruction in file "passeTypeRat.ml", line 84, characters 21-46
>   Called from Stdlib__List.map in file "list.ml", line 87, characters 15-19
>   Called from Rat__PasseTypeRat.analyse_type_bloc in file "passeTypeRat.ml" (inlined), line 146, characters 2-38
>   Called from Rat__PasseTypeRat.analyser in file "passeTypeRat.ml", line 169, characters 11-33
>   Called from Rat__Compilateur.Compilateur.analyser in file "compilateur.ml", line 34, characters 16-35
>   Called from Rat__Compilateur.compiler in file "compilateur.ml", line 96, characters 28-57
>   Called from Sans_fonction_type__Test.test in file "tests/type/sans_fonction/test.ml", line 398, characters 15-37
>   Re-raised at Sans_fonction_type__Test.test in file "tests/type/sans_fonction/test.ml", line 399, characters 59-66
>   Called from Sans_fonction_type__Test.test in file "tests/type/sans_fonction/test.ml", line 402, characters 4-16
>   Called from Sans_fonction_type__Test.test in file "tests/type/sans_fonction/test.ml", line 402, characters 4-16
>   Called from Sans_fonction_type__Test.(fun) in file "tests/type/sans_fonction/test.ml", line 408, characters 2-14
>
> FAILED 32 / 70 tests
[2]
